lib/logdna/client.rb

```diff
@@ -1,164 +1,130 @@
-require 'net/http'
-require 'socket'
-require 'json'
-require 'concurrent'
-require 'thread'
+# frozen_string_literal: true
+
+require "net/http"
+require "socket"
+require "json"
+require "concurrent"
+require "date"
 
 module Logdna
   class Client
-
     def initialize(request, uri, opts)
       @uri = uri
 
       # NOTE: buffer is in memory
-      @buffer = StringIO.new
-      @messages = []
-      @buffer_over_limit = false
+      @buffer = []
+      @buffer_byte_size = 0
 
-      @side_buffer = StringIO.new
       @side_messages = []
 
       @lock = Mutex.new
-      @task = nil
-
-      # NOTE: the byte limit only affects the message, not the entire message_hash
-      @actual_byte_limit = opts[:flushbyte] ||= Resources::FLUSH_BYTE_LIMIT
-      @actual_flush_interval = opts[:flushtime] ||= Resources::FLUSH_INTERVAL
-
-      @@request = request
-    end
-
-    def encode_message(msg)
-      msg = msg.to_s unless msg.instance_of? String
+      @flush_limit = opts[:flush_size] ? opts[:flush_size] : Resources::FLUSH_BYTE_LIMIT
+      @flush_interval = opts[:flush_interval] ? opts[:flush_interval] : Resources::FLUSH_INTERVAL
```

Comment:
Same thing here, I think you mistook my request. Use `||` instead of `||=`. That's the semantic intention.